Adding code snippet and screenshots of flexible job dependency (#9159)
circleci/circleci-docs: jekyll/_cci2/workflows.adoc

See the link:https://github.com/CircleCI-Public/circleci-demo-workflows/blob/sequential-branch-filter/.circleci/config.yml[Sample Sequential Workflow config] for a full example.

[#flexible-job-dependency]

> ==== Flexible job dependency

Reviewer: I think it would be good to move this section to the end of this sub section, so an h3 after the fan in-out section?

![Screenshot 2025-01-08 at 13 41 41](https://github.com/user-attachments/assets/3bb0aef1-8cc9-4b80-8b11-8239cd0f41e2)

Using `requires`, you can build complex dependency maps and fan in/out scenarios.

> This example shows a workflow with four sequential jobs. Each job waits until the "required" job attains a specified status.

Reviewer: ```suggestion
The following example is a configuration snippet of a workflow with four sequential jobs. Each job waits until the "required" job attains a specified status:
```

Reviewer: Just a suggestion for merging these two sentences

> This configuration snippet is an example of a workflow configured for flexible sequential jobs:
> 

Reviewer: ```suggestion
```

[source,yaml]
----
workflows:
  build-and-deploy:
    jobs:
      - build
      - deploy:
          requires:
            - build
      - release:
          requires:
            - deploy
      - rollback:
          requires:
            - release:
              - failed
              - canceled
----

> A job must wait until all upstream jobs in the dependency graph have attained either of the statuses specified under the corresponding `requires` key. In this example, the `deploy` job runs when the `build` job completes successfully:

Reviewer: ```suggestion
A job must wait until all upstream jobs in the dependency graph have attained one of the statuses specified under the corresponding `requires` key. In this example, the following dependencies are configured:
```

Reviewer: Just a suggestion to remove the duplication of the deploy job run explanation as it's in the bullet list too
